github/copilot.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// CopilotService provides access to the Copilot-related functions
+// in the GitHub API.
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/
+type CopilotService service
+
+// OrganizationCopilotDetails represents the details of an organization's Copilot for Business supbscription.
+type OrganizationCopilotDetails struct {
+	SeatBreakdown         *SeatBreakdown `json:"seat_breakdown"`
+	PublicCodeSuggestions string         `json:"public_code_suggestions"`
+	CopilotChat           string         `json:"copilot_chat"`
+	SeatManagementSetting string         `json:"seat_management_setting"`
+}
+
+// SeatBreakdown represents the breakdown of Copilot for Business seats for the organization.
+type SeatBreakdown struct {
+	Total               int64 `json:"total"`
+	AddedThisCycle      int64 `json:"added_this_cycle"`
+	PendingCancellation int64 `json:"pending_cancellation"`
+	PendingInvitation   int64 `json:"pending_invitation"`
+	ActiveThisCycle     int64 `json:"active_this_cycle"`
+	InactiveThisCycle   int64 `json:"inactive_this_cycle"`
+}
+
+type CopilotSeats struct {
+	TotalSeats int64                `json:"total_seats"`
+	Seats      []CopilotSeatDetails `json:"seats"`
+}
+
+// CopilotSeatDetails represents the details of a Copilot for Business seat.
+// Assignee can either be a User, Team, or Organization.
+type CopilotSeatDetails struct {
+	Assignee                interface{} `json:"assignee"`
```

Comment:
It may be ok to use `*User` here. There are a few other places where we use `*User` and let users distinguish using the `Type` field. @gmlewis will know what the preferred method is.

Reply:
That's a good point about Team being different. I think this might be the best way to handle it after all.